crates/bdk/src/wallet/mod.rs

```diff
@@ -200,6 +201,61 @@ pub enum InsertTxError {
 #[cfg(feature = "std")]
 impl<P: core::fmt::Display + core::fmt::Debug> std::error::Error for NewError<P> {}
 
+#[derive(Debug)]
+/// Error returned from [`TxBuilder::finish`]
+pub enum CreateTxError<P> {
+    /// There was a problem with the descriptors passed in
+    Descriptor(DescriptorError),
+    /// We were unable to write wallet data to the persistence backend
+    Persist(P),
+    /// There was a problem while extracting and manipulating policies
+    Policy(PolicyError),
+    /// TODO: replace this with specific error types
```

Comment:
I think it will be a good idea to remove the `CreateTxError::Bdk` branch in this PR as well. I think removing this will improve the API significantly (since the caller won't have to handle a large list of redundant error variants).

This can be our opportunity to avoid using `Error::Generic` and have "concrete" variants for timelock/tx-version errors.

If needed, we can have `CreateTxError` be a variant of `bdk::Error`, although I would avoid this if possible.

Reply:
If you like the general approach of adding a new `CreateTxError` enum but would rather that I replace the `Bbk` variant with the specific errors for this PR then I'm happy to do that. I just wanted to make sure I was on the right track before chasing down all the required variants.

Reply:
I should be able to get this knocked out today or tomorrow so we don't delay the alpha.1 too much more.

Reply:
So this took a bit longer than a day or two to finally wrap up, but should be all done now. 😅